### PR TITLE
Helm chart initial release

### PR DESCRIPTION
**Pull request by luigibk**
### JIRA link (if applicable) ###
 https://tools.hmcts.net/jira/browse/RPE-936

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -41,35 +39,6 @@ withPipeline(type, product, component) {
       env.IDAM_API_URL = 'https://preprod-idamapi.reform.hmcts.net:3511'
       env.IDAM_CLIENT_REDIRECT_URI = 'https://rpe-bulk-scan-processor-sandbox.service.core-compute-sandbox.internal/oauth2/callback'
 
-      def az = { cmd -> return sh(script: "env AZURE_CONFIG_DIR=/opt/jenkins/.azure-${env.SUBSCRIPTION_NAME} az $cmd", returnStdout: true).trim() }
```
Comment (timja) [inline Jenkinsfile_CNP]: 🎉

File: charts/bulk-scan-orchestrator/values.template.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      key: connectionString
+  environment:
+    POSTGRES_USER: hmcts
+    S2S_SECRET: "${S2S_SECRET}"
+    IDAM_CLIENT_SECRET: "${IDAM_CLIENT_SECRET}"
+    IDAM_USER_NAME: "${IDAM_USER_NAME}"
+    IDAM_USER_PASSWORD: "${IDAM_USER_PASSWORD}"
+    IDAM_USERS_BULKSCAN_USERNAME: "${IDAM_USER_NAME}"
+    IDAM_USERS_BULKSCAN_PASSWORD: "${IDAM_USER_PASSWORD}"
+    #ENVELOPES_QUEUE_CONNECTION_STRING: "$(SB_CONN_STRING);EntityPath=envelopes"
```
Comment (timja) [inline charts/bulk-scan-orchestrator/values.template.yaml]: remove? and same one line down

File: charts/bulk-scan-orchestrator/values.template.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  readinessPath: '/health'
+  livenessPath: '/health'
+  secrets:
+    SB_CONN_STRING:
+      secretRef: servicebus-secret-namespace-${SERVICE_NAME}
+      key: connectionString
+  environment:
+    S2S_SECRET: "${S2S_SECRET}"
+    IDAM_CLIENT_SECRET: "${IDAM_CLIENT_SECRET}"
+    IDAM_USER_NAME: "${IDAM_USER_NAME}"
```
Comment (timja) [inline charts/bulk-scan-orchestrator/values.template.yaml]: Is IDAM_USER_NAME and IDAM_USERS_BULKSCAN_USERNAME needed?

File: charts/bulk-scan-orchestrator/values.template.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    IDAM_USER_PASSWORD: "${IDAM_USER_PASSWORD}"
+    IDAM_USERS_BULKSCAN_USERNAME: "${IDAM_USER_NAME}"
```
Comment (timja) [inline charts/bulk-scan-orchestrator/values.template.yaml]: is the next PR adding properties volume and removing this 😉

File: charts/bulk-scan-orchestrator/values.template.yaml (hunk 3 above)
Comment (luigibk) [reply charts/bulk-scan-orchestrator/values.template.yaml]: It was when I wrote the k8s template.

File: charts/bulk-scan-orchestrator/values.template.yaml (hunk 4 above)
Comment (luigibk) [reply charts/bulk-scan-orchestrator/values.template.yaml]: Yes, I was waiting for that part to be tested ... no point in having to revert it

File: charts/bulk-scan-orchestrator/values.template.yaml (hunk 3 above)
Comment (timja) [reply charts/bulk-scan-orchestrator/values.template.yaml]: Its used in the functional tests not in the application, from a quick grep i just did

Comment (timja) [reply charts/bulk-scan-orchestrator/values.template.yaml]: i.e. IDAM_USER_NAME is tests only the bulkscan one is required

File: charts/bulk-scan-orchestrator/values.template.yaml [hunk 5]
```diff
@@ -9,12 +9,16 @@ java:
...
     IDAM_USERS_BULKSCAN_USERNAME: "${IDAM_USER_NAME}"
     IDAM_USERS_BULKSCAN_PASSWORD: "${IDAM_USER_PASSWORD}"
     ENVELOPES_QUEUE_CONNECTION_STRING: "$(SB_CONN_STRING);EntityPath=envelopes"
     PROCESSED_ENVELOPES_QUEUE_CONNECTION_STRING: "$(SB_CONN_STRING);EntityPath=processed-envelopes"
+  keyVaults:
+    "bulk-scan":
+      resourceGroup: bulk-scan
+      secrets:
+        - idam-client-secret
+        - microservicekey-bulk-scan-orchestrator
```
Comment (timja) [inline charts/bulk-scan-orchestrator/values.template.yaml]: shouldn't this come from the s2s vault?

Comment (luigibk) [reply charts/bulk-scan-orchestrator/values.template.yaml]: Well spotted! :-)

File: charts/bulk-scan-orchestrator/values.template.yaml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    ENVELOPES_QUEUE_CONNECTION_STRING: "$(SB_CONN_STRING);EntityPath=envelopes"
+    PROCESSED_ENVELOPES_QUEUE_CONNECTION_STRING: "$(SB_CONN_STRING);EntityPath=processed-envelopes"
+  keyVaults:
+    "bulk-scan":
+      resourceGroup: bulk-scan
+      secrets:
+        - idam-client-secret
+        - idam-users-bulkscan-username
+        - idam-users-bulkscan-password
+        - s2s-secret-for-tests
```
Comment (timja) [inline charts/bulk-scan-orchestrator/values.template.yaml]: this shouldn't be needed for the application to be run,
it should be:
`microservicekey-bulk-scan-orchestrator`
from the s2s vault

https://github.com/hmcts/bulk-scan-orchestrator/blob/master/infrastructure/main.tf#L112

File: charts/bulk-scan-orchestrator/values.template.yaml [hunk 7]
```diff
@@ -0,0 +1,37 @@
+java:
+  applicationPort: 8582
+  readinessPath: '/health'
+  livenessPath: '/health'
```
Comment (timja) [inline charts/bulk-scan-orchestrator/values.template.yaml]: why are these values set?
this line and the above

readiness is at the default and we don't want people overriding that for no reason, and adding a liveness check is adding a 1 line spring-boot starter

**Pull request merged**